### PR TITLE
Add more unit tests for redis quota adapter.

### PR DESCRIPTION
**Pull request by chowchow316**
The code coverage for redis quota adapter is about 70% for now, and most uncovered code are error handling part. 
This PR is trying to improve the code coverage by adding more test cases. Specifically, a fake server is used here to wrap miniredis. But still need to figure out how to cover the redis commands sent out line by line.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/608)

### CODE REVIEW

File: adapter/redisquota/redisquota_test.go [hunk 1]
```diff
@@ -421,3 +475,97 @@ func TestBadConfig(t *testing.T) {
...
+	}
+
+	b := newBuilder()
+	c := b.DefaultConfig().(*config.Params)
+	c.RedisServerUrl = f.Addr()
+	c.RateLimitAlgorithm = "fixed-window"
+	c.ConnectionPoolSize = 1
+	c.MinDeduplicationDuration = time.Duration(3600) * time.Second
+
+	a, _ := b.NewQuotasAspect(test.NewEnv(t), c, definitions)
```
Comment (douglas-reid) [inline adapter/redisquota/redisquota_test.go]: I think we might need to be checking for errors here:

```
--- FAIL: TestErrorResponse (0.00s)
	env.go:90: Could not create connection pool with redis: dial tcp [::1]:9736: getsockopt: connection refused
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x76ac33]

goroutine 90 [running]:
testing.tRunner.func1(0xc42023cb60)
	/home/jenkins/.cache/bazel/_bazel_jenkins/4889af1fce9f2f70c352cb6db3bd2e0b/external/golang_linux_amd64/src/testing/testing.go:622 +0x55f
panic(0x7ba720, 0x9b5990)
	/home/jenkins/.cache/bazel/_bazel_jenkins/4889af1fce9f2f70c352cb6db3bd2e0b/external/golang_linux_amd64/src/runtime/panic.go:489 +0x2f0
istio.io/mixer/adapter/redisquota.TestErrorResponse(0xc42023cb60)
	/home/jenkins/workspace/mixer/presubmit/src/istio.io/mixer/adapter/redisquota/redisquota_test.go:500 +0x7e3
testing.tRunner(0xc42023cb60, 0x81c0b0)
	/home/jenkins/.cache/bazel/_bazel_jenkins/4889af1fce9f2f70c352cb6db3bd2e0b/external/golang_linux_amd64/src/testing/testing.go:657 +0x108
created by testing.(*T).Run
	/home/jenkins/.cache/bazel/_bazel_jenkins/4889af1fce9f2f70c352cb6db3bd2e0b/external/golang_linux_amd64/src/testing/testing.go:697 +0x544
FAIL	istio.io/mixer/adapter/redisquota	0.376s
```

File: adapter/redisquota/redisquota_test.go [hunk 2]
```diff
@@ -421,3 +475,97 @@ func TestBadConfig(t *testing.T) {
 	}
 
 }
+
+func TestErrorResponse(t *testing.T) {
+	mockredis := testServer{errOnStart: false, errOnIntRet: true}
+	_, f, _ := mockredis.Start()
```
Comment (douglas-reid) [inline adapter/redisquota/redisquota_test.go]: this should probably be guarded. race condition testing seems to cause issues with multiple attempts to bind to the same port.

Comment (chowchow316) [inline adapter/redisquota/redisquota_test.go]: Thanks, working on it now.

File: adapter/redisquota/redisquota_test.go (hunk 1 above)
Comment (chowchow316) [inline adapter/redisquota/redisquota_test.go]: Just wondering where is this log? Did not see it via local racetest.

**Pull request merged**